### PR TITLE
[spring - formater z CI]

### PR DESCRIPTION
**Pull request by Bart3kTK**
- doadanie formatera mvn
- dodanie joba z:
	- sprawdzeniem formatu
	- akutalnie z mvn test (w przyszlosci deploy)

### CODE REVIEW

File: .github/workflows/spring_format_check.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: maven
+
+    - name: Spring format check
+      working-directory: ./spring
+      run: ./../scripts/spring/spring_format_check.sh
```
Comment (MaciejBinczarowski) [inline .github/workflows/spring_format_check.yml]: Put please new line at the end of file

File: scripts/spring/spring_format_check.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+git config --global user.name "github-actions"
+git config --global user.email "github-actions@github.com"
+
+if [ -n "$(git diff)" ]; then
+  echo "Code is not formatted properly. Please run 'mvn formatter:format'."
+  exit 1
+else
+  echo "No changes detected. Job run successfully!"
+  exit 0
+fi
```
Comment (MaciejBinczarowski) [inline scripts/spring/spring_format_check.sh]: new line

File: scripts/spring/spring_format_check.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+mvn formatter:format
+git config --global user.name "github-actions"
+git config --global user.email "github-actions@github.com"
+
+if [ -n "$(git diff)" ]; then
```
Comment (MaciejBinczarowski) [inline scripts/spring/spring_format_check.sh]: dont we have single command for checking format?

**Review by MaciejBinczarowski — CHANGES_REQUESTED** [review]
Put new lines in files. I also wonder if we should trigger build and test job after style check to avoid wasting computing time

File: scripts/spring/spring_format_check.sh (hunk 3 above)
Comment (Bart3kTK) [reply scripts/spring/spring_format_check.sh]: A okazało się że ma xd

File: scripts/spring/spring_format_check.sh (hunk 2 above)
Comment (Bart3kTK) [reply scripts/spring/spring_format_check.sh]: 🫨🫨

File: .github/workflows/spring_format_check.yml (hunk 1 above)
Comment (Bart3kTK) [reply .github/workflows/spring_format_check.yml]: 😵‍💫🫨😵‍💫

File: .github/workflows/spring_format_check.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+        cache: maven
+
+    - name: Spring format check
+      working-directory: ./spring
+      run: mvn formatter:validate
```
Comment (MaciejBinczarowski) [inline .github/workflows/spring_format_check.yml:29]: Great job!

**Review by MaciejBinczarowski — APPROVED** [review]

**Pull request merged**